src/ActionParameterHandler.ts

```diff
@@ -138,8 +138,41 @@ export class ActionParameterHandler {
                     value = this.transformValue(value, param);
                     value = this.validateValue(value, param); // note this one can return promise
                 }
+                return value;
+        }
+    }
+
+    /**
+     * Normalizes string value to number or boolean.
+     */
+    protected normalizeValue(value: any, type: string) {
+        switch (type) {
+            case "number":
+                if (value === "")
+                    return undefined;
```

Comment:
I think if a parameter is passed in the query with no value it should be casted to `null` instead of `undefined`.

I will add a normal comment to detail this.

Reply:
I've just extracted pleerock's code to separate method, I haven't changed nothing in normalizing logic.
https://github.com/pleerock/routing-controllers/blob/master/src/ActionParameterHandler.ts#L112

I agree that we should convert `""` to true for boolean, but for number we should use undefined if param value not exist or NaN if it's not a number string. However I would like to don't mix different things in one PR, as we first need to discuss the edge cases we should merge this `@QueryParams` PR and start a new one with better normalizing.